### PR TITLE
Fixed LateInitializationError

### LINKED ISSUE #508: Unhandled Exception: LateInitializationError: Field '_receivePort@1710301534' has not been initialized.
_Platform:_
- [ ]  Dart
- [X]  Flutter Android or iOS
- [ ]  Flutter Web

_IDE:_
- [ ]  VSCode
- [X]  IntelliJ/AS
- [ ]  XCode
- [ ]  Other, which one?

_split-debug-info and obfuscate (Flutter Android or iOS) or CanvasKit (Flutter Web):_
- [ ]  Enabled
- [X]  Disabled

_Platform installed with:_
- [X] pub.dev
- [ ] GitHub

Output of the command `flutter doctor -v` below:

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.2.2, on Microsoft Windows [Version 10.0.19042.867], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[√] Chrome - develop for the web
[!] Android Studio (not installed) // OBS: Android Studio is installed but not recognized by this command for some reason
[√] IntelliJ IDEA Ultimate Edition (version 2021.1)
[√] Connected device (3 available)
```

The version of the SDK (See pubspec.lock):
**5.1.0**

---
I have the following issue:

When I try to reinitizalize the Sdk with different parameters, I get the following error:
```
E/flutter (13798): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: LateInitializationError: Field '_receivePort@1710301534' has not been initialized.
E/flutter (13798): #0      IsolateErrorIntegration._receivePort (package:sentry/src/isolate_error_integration.dart)
E/flutter (13798): #1      IsolateErrorIntegration.close (package:sentry/src/isolate_error_integration.dart:26:5)
E/flutter (13798): #2      Hub.close (package:sentry/src/hub.dart:234:27)
E/flutter (13798): #3      Sentry._init (package:sentry/src/sentry.dart:104:17)
E/flutter (13798): <asynchronous suspension>
E/flutter (13798): #4      Sentry.init (package:sentry/src/sentry.dart:55:5)
E/flutter (13798): <asynchronous suspension>
E/flutter (13798): #5      SentryService._initializeClient (package:itini/core/services/sentry_service.dart:139:5)
E/flutter (13798): <asynchronous suspension>
```

**Steps to reproduce:**
1. Initialize SDK;
2. Close SDK;
3. Initialize SDK;

**Actual result:**
LateInitializationError is thrown.

**Expected result:**
No errors thrown.

### PR DESCRIPTION
**Pull request by ueman**
I somehow messed it up to push it to the other branch. I did however used the other contributors commits, so he does get credit.

## :scroll: Description
Updated the class `IsolatedErrorIntegration` to handle the case where the `close` method is called before the `_receivePort` being initialized.

## :bulb: Motivation and Context
I was seeing an error when reinitializing the Sentry SDK with different options.

Closes #508

## :green_heart: How did you test it?
I added tests.

## :pencil: Checklist

- [X] I reviewed submitted code
- [x] I added tests to verify changes
- [X] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: dart/test/initialization_test.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import 'package:sentry/sentry.dart';
+import 'package:test/test.dart';
+
+import 'mocks.dart';
+
+// Tests for the following issue
+// https://github.com/getsentry/sentry-dart/issues/508
+// There are no asserts, test are succesfull if no exceptions are thrown.
+void main() {
```
Comment (marandaneto) [inline dart/test/initialization_test.dart:11]: maybe set do a `setDown()`, nto sure if its called like this but the opposite of `setUp`, to call `Sentry.close` as well, so every test star in a clean state

**Comment by marandaneto** [comment]
@ueman please also check https://github.com/getsentry/sentry-dart/issues/508#issuecomment-868268197

**Comment by ueman** [comment]
> @ueman please also check [#508 (comment)](https://github.com/getsentry/sentry-dart/issues/508#issuecomment-868268197)

Done ✅

**Comment by marandaneto** [comment]
> > @ueman please also check [#508 (comment)](https://github.com/getsentry/sentry-dart/issues/508#issuecomment-868268197)
> 
> Done ✅

missing a `push` then?

**Comment by ueman** [comment]
> > > @ueman please also check [#508 (comment)](https://github.com/getsentry/sentry-dart/issues/508#issuecomment-868268197)
> > 
> > 
> > Done ✅
> 
> missing a `push` then?

No, it's the `tearDown` method which does it.

**Comment by marandaneto** [comment]
> > > > @ueman please also check [#508 (comment)](https://github.com/getsentry/sentry-dart/issues/508#issuecomment-868268197)
> > > 
> > > 
> > > Done ✅
> > 
> > 
> > missing a `push` then?
> 
> No, it's the `tearDown` method which does it.

nope, you're mistaking the comment, see:

> @ueman I believe NativeSdkIntegration can happen the same if close throws, _options might not be inited yet.
> can you check that too? thanks

cab be another PR as well, but the problem is rather the same

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**